### ISSUE TITLE
Refactor the workshop edit to use the _form partial

### ISSUE THREAD

**Issue by jabawack81** [codebar/planner#1118]
## Description of the issue 📄
Not a real issue but working on the issue #1104 I noticed that in the `edit` and `new` views for the workshops are using two different partials.

Comparing the two files I excluding `CSS` differences I can't see a reason for this difference:

<pre>
diff _edit_form.html.haml _form.html.haml
1c1
< = simple_form_for [:admin, @workshop], url: :admin_workshop<b>, method: :put,</b> do |f|
---
> = simple_form_for [:admin, @workshop], url: :admin_workshops do |f|
13c13
<       = f.input :host, as: :select, collection: Sponsor.all, include_blank: true, selected: (@workshop.host.id rescue '')
---
>       = f.input :host, as: :select<b>, required: true</b>, collection: Sponsor.all, include_blank: true, selected: (@workshop.host.id rescue '')
16c16
<       = f.association :sponsors
---
>       = f.association :sponsors<b>, required: true</b>
26,27c26,27
<       <b>%label Organisers</b>
<       <b>= f.collection_select :organisers, Member.all, :id, :full_name, { selected: @workshop.organisers.map(&:id) }, { multiple: true }</b>
---
>
>
</pre>

* using `PUT` on the `edit` form.
* marking `host` as required in the `edit` form
* marking `sponsors` as required in the `edit` form
* adding `organisers`  in the `edit` form

there is a reason behind this situation? if not I would like to open a PR (after my other PR, https://github.com/codebar/planner/pull/1108, is merged to avoid conflicts) to fix this.

Consideng the target odience of Codebar I think the planner could be the first webapp students look at so it should follow rails best practises as much as possible.

## Steps to fix 🛠

Update the `edit` view to use `_form.html.haml` as partial instead `_edit_form.html.haml`

## To do 📋

* [x] Claim this issue (comment below, or assign yourself if you are part of the codebar org)
* [x] Fork and clone the repository
* [x] Update the relevant files. Follow the steps to fix section in this issue.
* [x] Commit your changes as one commit. Use the title of this issue as your commit message
* [x] Submit a pull request
* [x] Mention this issue in the PR description by including it's number
* [ ] Have your pull request reviewed & merged by a codebar team member

**Comment by notapatch** [comment]
When creating a new workshop it doesn't validate/error out 

```
1. http://localhost:3000/admin/workshops/new
2. Fill in nothing (not the hardest step)
3. Press Save.

No error message. Nothing happens. (Tested Chrome and Firefox)

```
I would expect it to point out there's errors. That's definitely an issue which goes to show that duplicated code requires more management.

**Comment by jabawack81** [comment]
I didn't though to test the functionality of the page, thank you.

From what I can gather the form is not sent because is not valid (chapter, local_date, local_time and sponsors are marked as required) but no error message is displayed.

So I think could be a good idea to add that to the scope of this Issue and add display validation error.

**Comment by notapatch** [comment]
I played with it for a few minutes and couldn't work out the problem. I was going to put it up as a new issue but this issue was close enough. It would make sense to put them together (with the understanding that it's the maintainers last call). 

The investigation should cover the tests. A failed validation should have been tested.  I _think_ this is the one that should have covered it but otherwise dig out the test that does - either there is no test (add one) or the test doesn't work (change) ... either way I'm expecting new test code.

```
  # spec/features/admin/workshops_spec.rb  (or whatever was the right file)
  context 'creating a new workshop' do
    scenario 'successfully' do
     
     # happy path
     
    end
     
    #
    # sad path - why is this not doing it's job
    #
    scenario 'renders an error when no chapter has been selected' do
      visit new_admin_workshop_path

      fill_in 'Local date', with: Date.current
      fill_in 'Local time', with: '11:30'

      click_on 'Save'

      expect(page).to have_content('Chapter can\'t be blank')
    end
  end

```
The investigation should test all form validations are working. If there was a test change do other tests need changing? If it gets too much it's a new issue.

**Comment by jabawack81** [comment]
I've found the problem, the problem was the validastion from Foundation so I've disabled it, I've opened a PR https://github.com/codebar/planner/pull/1155

**Comment by despo** [comment]
Fixed as part of #1220 
https://github.com/codebar/planner/pull/1220/files#diff-397ff5ba4224791db024fa8afaceb131
